src/nettests/index.js

```diff
@@ -1,9 +1,106 @@
+import camelCase from 'camelcase'
+
+import percentage from '../cli/output/percentage'
+import wait from '../cli/output/wait'
+
+import {
+  Result,
+  Measurement
+} from '../config/db'
+
 import HTTPHeaderFieldManipulation from './http-header-field-manipulation'
 import HTTPInvalidRequestLine from './http-invalid-request-line'
 import WebConnectivity from './web-connectivity'
 import NDT from './ndt'
 import Base from './base'
 
+export const makeOoni = () => {
+  let dbOperations = [],
+      progress = null,
+      reportId = null,
+      uploaded = false,
+      localReportId = null,
+      isMk = false
+
+  let measurement = Measurement.build({
+    state: 'active'
+  })
+
+  const init = (nt) => {
+    // XXX we use duck typing to see if the argument is a measurement Kit
+    // nettest. Maybe we should change this.
+    if (nt.test && nt.name && nt.setOptions) isMk = true
+
+    measurement.name = camelCase(nt.name)
+    measurement.reportFile = measurement.makeReportFile()
+    dbOperations.push(measurement.save())
+
+    if (isMk) {
+      nt.test.set_options('no_file_report', '0')
+      nt.test.set_output_filepath(measurement.reportFile)
+      nt.on('log', (severity, message) => {
+        // XXX this a workaround due to a bug in MK
+        // I actually also need to know if the report has been created
+        if (message.startsWith('Report ID:') && reportId === null) {
+          reportId = message.split(':')[1].trim()
+          uploaded = true
+        }
+      })
+      nt.on('entry', entry => {
+        // XXX This is a bit of a hack
+        // When we don't have a reportId with the collector we set the
+        // localReportId to the first measurement id.
+        // When I have proper ways of handling this:
+        // * https://github.com/measurement-kit/measurement-kit/issues/1469
+        // * https://github.com/measurement-kit/measurement-kit/issues/1462
+        // This should be a bit cleaner
+        if (reportId === null) {
+          uploaded = false
+          reportId = `LOCAL-${entry.id}`
+        }
+        if (!measurement.country) {
+          dbOperations.push(measurement.update({
+            reportId: reportId,
+            country: entry['probe_cc'],
+            asn: entry['probe_asn'],
+            ip: entry['ip'],
+            // We append the Z to make moment understand it's UTC
```

Comment:
@hellais yeah, I think we should really consider somehow adding `Z` always.

Reply:
true, but this needs a spec change

Reply:
Yep. May I suggest that we create an issue for that?